esx_service/vmdk_ops.py

```diff
@@ -794,78 +851,57 @@ def disk_attach(vmdk_path, vm):
         pvsci = [d for d in controllers
                    if type(d) == vim.ParaVirtualSCSIController and
                       d.key == device.controllerKey]
+        
         return dev_info(device.unitNumber,
-                        get_controller_pci_slot(vm, pvsci,
+                        get_controller_pci_slot(vm, pvsci[0],
                                                 offset_from_bus_number))
+        
 
     # Disk isn't attached, make sure we have a PVSCI and add it if we don't
-    # TODO: add more controllers if we are out of slots. Issue #38
-
     # check if we already have a pvsci one
     pvsci = [d for d in controllers
              if type(d) == vim.ParaVirtualSCSIController]
+    disk_slot = None         
     if len(pvsci) > 0:
-        disk_slot = None  # need to find out
-        controller_key = pvsci[0].key
-        pci_slot_number = get_controller_pci_slot(vm, pvsci,
-                                                  offset_from_bus_number)
-    else:
-        logging.warning(
-            "Warning: PVSCI adapter is missing - trying to add one...")
+        idx, disk_slot = find_available_disk_slot(vm, devices, pvsci, offset_from_bus_number);
+        if (disk_slot is not None):
```

Comment:
I think we still need the check (disk_slot it not None). If we change the check to if (disk_slot), then when "disk_slot" is 0, then the code will not work